programs/mango-v4/src/state/bank.rs

```diff
@@ -557,10 +577,23 @@ impl Bank {
         allow_dusting: bool,
         now_ts: u64,
     ) -> Result<bool> {
-        let opening_indexed_position = position.indexed_position;
-        let result = self.deposit_internal(position, native_amount, allow_dusting, now_ts)?;
-        self.update_cumulative_interest(position, opening_indexed_position);
-        Ok(result)
+        if position.allow_lending() {
+            assert!(position.unlendable_deposits == 0);
+            let opening_indexed_position = position.indexed_position;
+            let result = self.deposit_internal(position, native_amount, allow_dusting, now_ts)?;
```

Comment:
I would rename the `deposit_internal` method to make it explicit that it's only for "lendable position"